### PR TITLE
Remove search module, porting query stuff to fetch

### PR DESCRIPTION
**Pull request by jcrossley3**

### CODE REVIEW

File: modules/ingestor/src/endpoints/advisory.rs [hunk 1]
```diff
@@ -37,27 +37,6 @@ pub async fn upload_advisory(
     Ok(HttpResponse::Created().json(advisory_id))
 }
 
-#[utoipa::path(
```
Comment (ctron) [inline modules/ingestor/src/endpoints/advisory.rs:40]: I think it's ok to remove it from here, but it should end up somewhere else. I can't find that, did I overlook it?

**Review by ctron — CHANGES_REQUESTED** [review]
It looks to me like the download (sbom, advisories) endpoint have simply been dropped. If that's the case, I am not ok with that.

File: modules/ingestor/src/endpoints/advisory.rs (hunk 1 above)
Comment (jcrossley3) [reply modules/ingestor/src/endpoints/advisory.rs:40]: It's [already in the fetch module](https://github.com/trustification/trustify/blob/bc2c73fde8fd0b433c9facd784afb556846f5dee/modules/fetch/src/endpoints/advisory.rs#L37-L51), so not a part of this PR

File: modules/ingestor/src/endpoints/sbom.rs [hunk 2]
```diff
@@ -30,24 +30,3 @@ pub async fn upload_sbom(
...
-#[utoipa::path(
-    tag = "ingestor",
-    responses(
-        (status = 200, description = "Download a an SBOM", body = Vec<u8>),
-        (status = 404, description = "The document could not be found"),
-    )
-)]
-#[get("/sboms/{id}")]
-/// Download an SBOM
-pub async fn download_sbom(
```
[the remark this answers is not in the document]
Comment (jcrossley3) [reply modules/ingestor/src/endpoints/sbom.rs:43]: I thought this was in the fetch module, too, but I don't see it now. :)

I think it makes sense for it to live there, I think, if we're ok with keeping all the GET's in fetch.

File: modules/ingestor/src/endpoints/advisory.rs (hunk 1 above)
Comment (ctron) [reply modules/ingestor/src/endpoints/advisory.rs:40]: Yea, there's a version with the digest, which I am not sure we use much. I think it makes sense having both versions with the `id`.

Comment (jcrossley3) [reply modules/ingestor/src/endpoints/advisory.rs:40]: I'll leave these in for now, but we need to agree on what `xxx` is in this request: `/api/v1/advisory/{xxx}`. Is it the entity's `id` field or its `sha256` field? Or do we need an api for both?

Complicating matters further, these will also work:

- `/api/v1/advisory?q=id%3Dfoo`
- `/api/v1/advisory?q=sha256%3Dfoo`

Comment (ctron) [reply modules/ingestor/src/endpoints/advisory.rs]: I would go for the `id`.

Comment (bobmcwhirter) [reply modules/ingestor/src/endpoints/advisory.rs]: But we'd prefer to not leak IDs. And sha can he deterministic.

**Review by ctron — APPROVED** [review]

File: modules/ingestor/src/endpoints/advisory.rs (hunk 1 above)
Comment (jcrossley3) [reply modules/ingestor/src/endpoints/advisory.rs]: Yeah, I thought we agreed IDs wouldn't be a part of the public API.

Potentially relevant: #157 #158

**Pull request merged**